chapter01/1.3 - URLify/urlify.js

```diff
@@ -0,0 +1,24 @@
+var urlify = function(str, length) {
+  // have a pointer to check from start to end
+  var strArr = str.split('');
+  var pointer = 0;
+  while (pointer < str.length) {
+    if (strArr[pointer] === ' ') {
+      // *** needs more work here, a little wierd
+      // not handling trailing spaces properly
+      for (var i = str.length - 1; i > pointer + 3; i--) {
+        strArr[i] = str[i - 2];
+      }
+      strArr[pointer] = '%';
+      strArr[pointer+1] = '2';
```

Comment:
Let me clarify what I mean by "relevant algorithms" with an example. In a solution to question 1.1 "Is Unique" (pg. 90), it would be okay to use your language's built-in hash table implementation because in asking this question, an interviewer would not be trying to discern whether or not you know how to implement a hash table, but rather that you know when to use appropriate data structures. On the other hand, in a solution to question 7.12 "Hash Table" (pg. 129), it would not be okay to simply use your language's built-in hash table implementation because the point of the question is to test whether you know how to implement one from scratch. An explicit algorithm would be required for question 7.12 which of course would not be idiomatic to the language but required nonetheless.